tests/apple-exporter/components/test-class-audio.php

```diff
@@ -18,6 +18,38 @@ public function testGeneratedJSON() {
 		$this->assertEquals( 'http://someurl.com/audio-file.mp3?some_query=string', $json['URL'] );
 	}
 
+	/**
+	 * Tests HTML formatting with captions.
+	 *
+	 * @access public
+	 */
+	public function testCaption() {
+		$workspace = $this->prophet->prophesize( '\Exporter\Workspace' );
+
+		// Pass the mock workspace as a dependency
+		$component = new Audio( '<figure class="wp-block-audio"><audio controls="" src="https://www.soundhelix.com/examples/mp3/SoundHelix-Song-1.mp3"/><figcaption>caption</figcaption></figure>',
+			$workspace->reveal(), $this->settings, $this->styles, $this->layouts );
+
+		// Test.
+		$this->assertEquals(
+			array(
+				'role' => 'container',
+				'components' => array(
+					array(
+						"role" => "audio",
+						"URL" => "https://www.soundhelix.com/examples/mp3/SoundHelix-Song-1.mp3",
```

Comment:
We'll need a way to run this test without actually running the `get_headers` call and having the unit test check this URL. Something like this:
```php
namespace Apple_Exporter\Components;
class MockAudio extends Audio {
    protected static function remote_file_exists( $node ) {
        return true;
    }
}
```

Then, instead of doing `new Audio` you would do `new MockAudio` and the test should pass without performing the external call.

You could put the mock class in `tests/mocks/class-mock-audio.php` and then `require_once` it at the top of this file.